aten/src/ATen/native/hammerblade/Resize.h

```diff
@@ -1,34 +1,38 @@
 #pragma once
 
 #include <ATen/ATen.h>
-#include <TH/THTensor.hpp>
+#include <ATen/native/hammerblade/HammerBladeTensor.h>
 #include <ATen/hammerblade/HammerBladeContext.h>
 #include <ATen/native/hammerblade/Offload.h>
 
 namespace at { namespace native {
 
-static inline void HBStorage_resize(THStorage* storage, ptrdiff_t size) {
+static inline void hammerblade_resize(c10::StorageImpl* storage, ptrdiff_t size) {
   if (storage->resizable()) {
     /* case when the allocator does not have a realloc defined */
+
+    int device = c10::hammerblade::current_device();
     at::DataPtr new_data;
     if (size != 0) {
       new_data = storage->allocator()->allocate(storage->itemsize() * size);
+    } else {
+      new_data = at::DataPtr(nullptr, at::Device(at::DeviceType::HAMMERBLADE, device));
```

Comment:
This is kind of important. Even if a tensor has 0 size, we still need to make sure it lives on hammerblade.

Reply:
I might be missing something here... if the size is 0, there's nothing related this tensor on the device, right? And, the metadata `tensor.device` would be on the host, correct?

Reply:
You are right in terms that if the size is 0, there is nothing on device. However, we still want to mark this tensor as a HammerBlade tensor. The way to do so is to have a HammerBlade DataPtr. ;)